app/cli/internal/action/artifact_download.go

```diff
@@ -60,26 +62,36 @@ func (a *ArtifactDownload) Run(downloadPath, digest string) error {
 	}
 
 	if downloadPath == "" {
-		var err error
 		downloadPath, err = os.Getwd()
 		if err != nil {
 			return fmt.Errorf("getting current dir: %w", err)
 		}
 	}
 
-	downloadPath = path.Join(downloadPath, info.Filename)
-	f, err := os.Create(downloadPath)
-	if err != nil {
-		return fmt.Errorf("creating destination file: %w", err)
+	// Determine output destination
+	outputPath := path.Join(downloadPath, info.Filename)
```

Comment:
I'd make the use of `downloadPath` incompatible with providing the file. To me, the output is not the name of the file but the full path. 

So basically, if you provide a file, that's what we use, otherwise the system will work as before.

What do you think?

Reply:
I had mixed feelings when implementing this and felt the same way as you. Going to do the changes, I believe it's more natural that way!